File: Brainiacs/src/main/java/brainiacs/games/GameService.java
package brainiacs.games;

import java.util.ArrayList;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import brainiacs.courses.Course;

@Service
public class GameService
{
	
	@Autowired
	private GameRepository gameRepo;
	@Autowired
	private QuestionRepository qRepo;
	
	public void addGame(Game game)
	{
		gameRepo.save(game);
	}
	
	
	public ArrayList<Game> getGamesByCourseName(String name)
	{
		ArrayList<Game> games = new ArrayList<Game>();
		gameRepo.getGamesByCourseName(name).forEach(games::add);
		return games;
	}
	
	public ArrayList<Game> getGames()
	{
		
		ArrayList<Game> games = new ArrayList<Game>();
		gameRepo.findAll().forEach(games::add);
		return games;	
	}
	
	public Game getGame(Integer id)
	{
		return gameRepo.findOne(id);
	}
	
}

File: BrainiacsFront/js/services/GameService.js
var GameService = angular.module("GameService", []);

GameService.service("Game", function(){
   
    var game =
    {
        questions: new Array(),
        course: null,
        name: null,
        score: 0
    }

    
    return{
        getGame: function()
        {
            return game;
        },
        setGame: function(g)
        {
            game = g;
        },
        addQuestion: function(question)
        {
            game.questions.push(question);
        },
        setCourse: function(c)
        {
            game.course = c;
        },
        setName: function(name)
        {
            game.name = name;
        }
        
    }
    
});